client-state-machine.go

```diff
@@ -128,6 +146,18 @@ func (state clientStateStart) Next(hr handshakeMessageReader) (HandshakeState, [
 			return nil, nil, AlertInternalError
 		}
 	}
+	if cct != nil {
```

Comment:
Why did you split up the extension construction? it seems like you could just have this at the end of the cct, sct construction block.

Reply:
Parallelism with how ALPN works already.  Refactoring both.